demos/blockfactory/code_to_block_xml.js

```diff
@@ -0,0 +1,393 @@
+  block1.append(newNode('mutation', {types:types.length}))
+  for(let i=0; i<types.length; i++){
+    let type = types[i]
+    let value = newNode('value', {name:'TYPE'+i})
+    block1.append(value)
+    data.dst.current = value
+    parseType(data, type)
+  }
+  data.dst.current = block1  
+  return 0
+}
+
+//<xml xmlns="http://www.w3.org/1999/xhtml">
+//  <block type="factory_base" id="@M]TSC%Vc,f=XNfsm,:]" x="0" y="0">
+//    <mutation connections="NONE"></mutation>
+//    <field name="NAME">block_type</field>
+//    <field name="INLINE">AUTO</field>
+//    <field name="CONNECTIONS">NONE</field>
+//    <statement name="INPUTS">
+//      <block type="input_value" id="Z`L[c)W,AG6CA!.nO4}t">
+//        <field name="INPUTNAME">NAME2</field>
+//        <field name="ALIGN">LEFT</field>
+//        <statement name="FIELDS">
+//          <block type="field_input" id="7*_1AO9o^1*i0O~8Rux#">
+//            <field name="TEXT">default</field>
+//            <field name="FIELDNAME">NAME3</field>
+//          </block>
+//        </statement>
+//        <value name="TYPE">
+//          <block type="type_group" id="X`wg0~`2k/LSjF*A)0[g">
+//            <mutation types="3"></mutation>
+//            <value name="TYPE0">
+//              <block type="type_boolean" id="6Ig;!bzV3$F3wbeCX0|."></block>
+//            </value>
+//            <value name="TYPE1">
+//              <block type="type_number" id=".8=Fy(X4Y)^AxL=2`|^k"></block>
+//            </value>
+//            <value name="TYPE2">
+//              <block type="type_other" id="xZ#E~g4$mU9EW[x^F*L^">
+//                <field name="TYPE">abc</field>
+//              </block>
+//            </value>
+//          </block>
+//        </value>
+//        <next>
+//          <block type="input_statement" id=",3WU]2u2yr*T=]Uf~df)">
+//            <field name="INPUTNAME">NAME</field>
+//            <field name="ALIGN">LEFT</field>
+//            <statement name="FIELDS">
+//              <block type="field_input" id="J=JfkI!1B_4+R3kcLKPQ">
+//                <field name="TEXT">default</field>
+//                <field name="FIELDNAME">NAME4</field>
+//              </block>
+//            </statement>
+//            <value name="TYPE">
+//              <block type="type_string" id=",0DAA|CKXhak(}mF@+;+"></block>
+//            </value>
+//          </block>
+//        </next>
+//      </block>
+//    </statement>
+//    <value name="TOOLTIP">
+//      <block type="text" id="Z|jYwov.`iDkT8i%JpY4">
+//        <field name="TEXT"></field>
+//      </block>
+//    </value>
+//    <value name="HELPURL">
+//      <block type="text" id="U-I4}8m|9f|z{EG.oM)_">
+//        <field name="TEXT"></field>
+//      </block>
+//    </value>
+//    <value name="COLOUR">
+//      <block type="colour_hue" id="v51i_=]YEpM~yh84R/9[">
+//        <mutation colour="#5b61a5"></mutation>
+//        <field name="HUE">235</field>
+//      </block>
+//    </value>
+//  </block>
+//</xml>
+
+var parseType = function(data, type){

```

Comment:
Several of these look like they would be a better fit under factory_utils.js.
Please add documentation when doing so.